lib/src/types.dart

```diff
@@ -9,100 +9,9 @@
   5. add identifying key to PostgreSQLFormat.dataTypeStringForDataType
  */
 
-/// Supported data types.
-enum PostgreSQLDataType {
-  /// Must be a [String].
-  text,
+import 'package:postgres/src/v3/types.dart';
 
-  /// Must be an [int] (4-byte integer)
-  integer,
-
-  /// Must be an [int] (2-byte integer)
-  smallInteger,
-
-  /// Must be an [int] (8-byte integer)
-  bigInteger,
-
-  /// Must be an [int] (autoincrementing 4-byte integer)
-  serial,
-
-  /// Must be an [int] (autoincrementing 8-byte integer)
-  bigSerial,
-
-  /// Must be a [double] (32-bit floating point value)
-  real,
-
-  /// Must be a [double] (64-bit floating point value)
-  double,
-
-  /// Must be a [bool]
-  boolean,
-
-  /// Must be a [DateTime] (microsecond date and time precision)
-  timestampWithoutTimezone,
-
-  /// Must be a [DateTime] (microsecond date and time precision)
-  timestampWithTimezone,
-
-  /// Must be a [Duration]
-  interval,
-
-  /// Must be a [List<int>]
-  numeric,
-
-  /// Must be a [DateTime] (contains year, month and day only)
-  date,
-
-  /// Must be encodable via [json.encode].
-  ///
-  /// Values will be encoded via [json.encode] before being sent to the database.
-  jsonb,
-
-  /// Must be encodable via [json.encode].
-  ///
-  /// Values will be encoded via [json.encode] before being sent to the database.
-  json,
-
-  /// Must be a [List] of [int].
-  ///
-  /// Each element of the list must fit into a byte (0-255).
-  byteArray,
-
-  /// Must be a [String]
-  ///
-  /// Used for internal pg structure names
-  name,
-
-  /// Must be a [String].
-  ///
-  /// Must contain 32 hexadecimal characters. May contain any number of '-' characters.
-  /// When returned from database, format will be xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.
-  uuid,
-
-  /// Must be a [PgPoint]
-  point,
-
-  /// Must be a [List<bool>]
-  booleanArray,
-
-  /// Must be a [List<int>]
-  integerArray,
-
-  /// Must be a [List<String>]
-  textArray,
-
-  /// Must be a [List<double>]
-  doubleArray,
-
-  /// Must be a [String]
-  varChar,
-
-  /// Must be a [List<String>]
-  varCharArray,
-
-  /// Must be a [List] of encodable objects
-  jsonbArray,
-}
+typedef PostgreSQLDataType = PgDataType<Object>;
```

Comment:
I haven't seen such typedef in the wild, but I'll be happy if this is backward-compatible.

However, we can also consider using a separate `PgDataType` class (not enum), if we seem to need more than what the enum allows. Not sure though, if this non-enum class would be needed...

Reply:
I think it should be fully backwards compatible. Now that we have enums with methods and fields in Dart, I hope the enum class will work :crossed_fingers: Making it an enum has the benefit of compiler-checked exhaustive switch statements.